External/CSG/Classes/CSG_Model.cs

```diff
@@ -0,0 +1,139 @@
+using System;
+using UnityEngine;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace UnityEngine.ProBuilder.Csg
+{
+    /// <summary>
+    /// Representation of a mesh in CSG terms. Contains methods for translating to and from UnityEngine.Mesh.
+    /// </summary>
+    sealed class CSG_Model
+    {
+        internal List<CSG_Vertex> vertices;
+        List<Material> m_Materials;
+        List<List<int>> m_Indices;
+
+        public List<Material> materials
+        {
+            get { return m_Materials; }
+            set { m_Materials = value; }
+        }
+
+        public Mesh mesh
+        {
+            get { return (Mesh)this; }
+        }
+
+        public CSG_Model()
```

Comment:
I fail to see the point of this public constructor as we can't set the indices from outside (additionnal, m_Material would be null after instantiation).
The only ways I see to have an instance of CSG_Model with valid data is to use the constructor overloads.